client/src/controllers/KeyboardController.ts

```diff
@@ -0,0 +1,39 @@
+import { Controller } from '@hotwired/stimulus';
+import Mousetrap from 'mousetrap';
+
+/**
+ * Adds the ability to trigger a button click event using a keyboard shortcut
+ * declared on the controlled element.
+ *
+ * @example
+ * ```html
+ * <button type="button" data-controller="w-kbd" data-w-kbd-key="[">Trigger me with the <kbd>[</kbd> key.</button>
+ * ```
+ */
+export class KeyboardController extends Controller<HTMLButtonElement> {
+  static values = { key: { default: '', type: String } };
+
```

Comment:
You need to tell TypeScript that the value exists.  

```suggestion

  declare keyValue: string;

```

https://stimulus.hotwired.dev/reference/using-typescript

Reply:
Thank you for this! I was stuck the past 2 days on it 😅

Reply:
All good. It's a lot to learn I know. At least the experience will be strong in your memory for the next time 😅

It's hard to know what will be confusing for new devs, one tip I'd suggest is if you get stuck like that again, look at some other similar code it might help. 

But of course always feel free to ask questions like you have been. We'll always try to help if we can.

Reply:
Thanks for the tip! I did look at other codes but unfortunately, they were js implementations, not ts ones 😅. I'll keep this in mind next time 🙌